doc/writing_modules.rst

```diff
@@ -27,6 +27,34 @@ which if you are used to XDG_CONFIG paths relates to:
 You can also specify the modules location using ``py3status -i <path to custom
 modules directory>`` in your i3 configuration file.
 
+Publish your custom modules on PyPI
+-----------------------------------
+
+.. note::
+    TODO add a remark here in which version this functionality was added
+
+To share custom modules that are not directly part of py3status you can create
+a package and publish it to PyPI.
+
+If you have installed py3status in a virtualenv (maybe because your custom module
+has dependencies that need to be available) you can also create an installable
+package from your module and publish it on PyPI.
+
+py3status can discover custom modules if they are installed in the same host interpreter
+and if an entry_point in the package ``setup.py`` is defined::
+
+    setup(
+        entry_points={"py3status": ["module = path.to.py3status_module"]},
+    )
+
+Although it is possible to install packages from PYPi in the distributed Python
+interpreter it is recommended to install py3status into a virtualenv and install
```

Comment:
I'm not so sure about advocating people to install py3status in a virtualenv. This targets mostly "advanced" users because people expect the `py3status` command to be available on their system.

I don't want to start dealing with ppl complaining to not find their py3status "bin" or i3wm startup problems due to missing PATHs.

Reply:
True. That's not what I was trying to say anyway. I would say that creating an installable py3status package that is registered via entry point is already an advanced usage and in that advanced usage scenario it then makes sense to have py3status running in a virtualenv to avoid cluttering your host interpreter with depdendencies. I'll see if I can rewrite that with a warning to make this clearer. Will be very busy though during the next two weeks, but after that I can pick things up here and finish them.

Reply:
Maybe gimme write access to your fork, so I can try working on the phrasing mate?

Reply:
Sure - just added you as a collaborator there, but I actually thought that allowing changes from maintainers would take care of that but seems not to be the case.

Reply:
> I actually thought that allowing changes from maintainers would take care of that but seems not to be the case.

I'm sorry I'm not sure I understand?

Reply:
I meant this: https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Reply:
oohh first time I hear about that so I'm apologize I didn't even try before asking mate!